docs/limitrange.md

```diff
@@ -0,0 +1,104 @@
+-   container2 : 2 CPU, 250m memory
+-   container3 : 3 CPU, 500m memory
+
+The computation will be:
+-   CPU : 2 (max init containers) + 6 (sum of containers) = 8 CPU
+-   Memory: 200m (max init containers) + 800m (sum of containers) = 1000m (1G)
+
+## Tekton support
+
+The way Limits and Requests works in Kubernetes is because it is assumed that all containers run in parallel, and init container run before, each one after the others.
+
+That assumption — containers running in parallel — is not true in Tekton. They do all start together (because there is no way around this) **but** the *entrypoint hack* is making sure they actually run in sequence and thus there is always only one container that is actually consuming some resource at the same time.
+
+This means, we need to handle limits, request and LimitRanges in a *non-standard* way. Let's try to define that. Tekton needs to take into account all the aspect of the LimitRange : the min/max as well as the default. If there is no default, but there is min/max, Tekton need then to **set** a default value that is between the min/max. If we set the value too low, the Pod won't be able to be created, similar if we set the value too high. **But** those values are set on **containers**, so we **have to** do our own computation to know what request to put on each containers.
+
+
+## A LimitRange is in the namespace
+
+We need to get the default (limits), default requests, min and max values (if they are here).
+One thing to note is that, in the case of a LimitRange being present, we need to **not rely** on the pod mutation webhook that takes the default into account ; what this means is, we need to specify all request and limits ourselves so that the mutation webhook doesn't have any work to do.
+
+- **No default value:** if there is no default value, we need to treat the min as the
+  default. I think that's also what k8s does, at least in our computation.
+- **Default value:** we need to "try" to respect that as much as possible.
+  - `defaultLimit` but no `defaultRequest`, then we set `defaultRequest` to be same as `defaultLimit`.
+  - `defaultRequest` but no `defaultlimit`, then we use the `min` limit as the `defaultLimit`
+  - no `defaultLimit`, no `defaultRequest`, then we use the `min` limit (or request) as
+    `defaultLimit` (and `defaultRequest`).
+
+Now for the container's computations, here are the rules
+- **init containers:** they won't be summed, so the rules are simple
+  - a container needs to have request and limits at least at the `min` and set to the `default` if any.
+  - *use the default requests and the default limits (coming from the defaultLimit, or the min, …)*
+- **containers:** those will be summed at the end, so it gets a bit complex
+  - a container needs to have request and limits at least at the `min`
+  - the sum of the container request/limits **should be** as small as possible. This should be
+    ensured by using the "smallest" possible request on it.
```

Comment:
maybe this answers my question above about why the `min` is being used as the `defaultLimit` - i think it would help me understand better if you explained near the beginning of the section that we want the sum to be as small as possible and why

Reply:
i think this was updated - and/or my understanding is just better now, but anyway lgtm 👍 👍 👍